### PR TITLE
[WIP] Moving back to master

### PR DESCRIPTION
**Pull request by mcdonnelldean**
Hey @mirceaalexandru I've set up the pull request for this so I can get some changes before it's merged in. I'll take a look then when you have it updated. Cheers

### CODE REVIEW

File: config/config.example.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+module.exports = {
+  server: {
+    port: 3050,
+  },
+
+  // DB connections
+  "db": {
```
Comment (mcdonnelldean) [inline config/config.example.js]: We will need a Dockerfile for whatever storage we use.

File: server/plugins/concorda-user/controller/application.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+module.exports = function () {
+  var seneca = this
+  var name = 'applicationCtrl'
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: aplicationCtrl -> concorda

File: server/plugins/concorda-user/controller/application.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      apps = apps || []
+      for (var i in apps){
+        apps[i] = apps[i].data$(false)
+        response(null, { data: apps })
+      }
+    })
+  }
+
+  function create( msg, response ) {
+    this.make$('application', msg.data).save$(function(err, app){
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: 'application' -> 'connected_service'

File: server/plugins/concorda-user/controller/application.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      console.log(app)
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: Log to seneca

File: server/plugins/concorda-user/controller/application.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      response(null, { data: app.data$(false) })
+    })
+  }
+
+  seneca
+    .add( {role: name, cmd: 'list'}, list )
+    .add( {role: name, cmd: 'create'}, create )
+
+  seneca.act( {role: 'web', use: {
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: remove this for now, we will manually wire up, until we are sure what we need I don't want to support it :D

File: server/plugins/concorda-user/controller/application.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    prefix: '/api',
+    pin: {role: name, cmd: '*'},
+    map: {
+      list: { GET: true, alias: 'application'},
+      create: { POST: true, data: true, alias: 'application'}
+    }
+  }} )
+
+  return {
+    name: 'concorda-user-application'
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: 'concorda-user-application' -> options.name

**Comment by mirceaalexandru** [comment]
@dean thanks this PR was the first thing I wanted to do tomorrow morning :)

File: server/plugins/concorda-user/controller/application.js (hunk 2 above)
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: name should me in default options like

```
var options = {
  name: ''
}

options = seneca.util.extend(options, opts) 
```

File: server/plugins/concorda-user/controller/application.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+module.exports = function () {
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: add opts to allow options in

File: server/plugins/mongo-db/app.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+'use strict'
```
Comment (mcdonnelldean) [inline server/plugins/mongo-db/app.js]: Don't need this file, we don't care where they store it. It's up to the server to tell us. Can be removed

File: server/concorda.js [hunk 9]
```diff
@@ -3,6 +3,10 @@ var Package = require('../package.json')
 var SenecaUser = require('seneca-user')
 var SenecaAuth = require('seneca-auth')
 
+// load plugins
+var MongoDB = require('../server/plugins/mongo-db/app')
```
Comment (mcdonnelldean) [inline server/concorda.js]: Remove with file (you can add mongo in the hapi server where we set up seneca)

File: server/concorda.js [hunk 10]
```diff
@@ -3,6 +3,10 @@ var Package = require('../package.json')
...
+var ConcordaUser = require('../server/plugins/concorda-user/app')
```
Comment (mcdonnelldean) [inline server/concorda.js:8]: merge all other files into /server/concorda-user.js

File: server/plugins/concorda-user/app.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
+'use strict'
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/app.js:1]: Don't need. We are going to do everything in one file for now.

File: server/plugins/concorda-user/controller/application.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+  var seneca = this
+  var name = 'applicationCtrl'
+
+  function list( msg, response ) {
+    this.make$('application').list$({}, function(err, apps){
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: 'application' -> 'connected_service'

**Comment by mcdonnelldean** [comment]
@mirceaalexandru Ok review done. It all looks pretty much solid, super job on the tests. 

Mostly just name changes and removing some things that aren't needed. Otherwise, good job, it's all looking great.

File: package.json [hunk 13]
```diff
@@ -53,9 +54,9 @@
     "redux-logger": "2.3.x",
     "redux-simple-router": "1.0.x",
     "redux-thunk": "1.0.x",
-    "seneca-auth": "https://github.com/mirceaalexandru/seneca-auth#hapi-bell",
+    "seneca-auth": "mirceaalexandru/seneca-auth#hapi-express",
     "seneca-user": "0.2.x",
-    "seneca-web": "0.5.0",
+    "seneca-web": "mirceaalexandru/seneca-web",
```
Comment (mcdonnelldean) [inline package.json]: @geek We need to look at bringing this in the seneca, thoughts?

File: server/plugins/concorda-user/controller/application.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+      if (err){
+        return response(err)
+      }
+
+      apps = apps || []
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: apps > connectedServices

File: server/plugins/concorda-user/controller/application.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+      for (var i in apps){
+        apps[i] = apps[i].data$(false)
+        response(null, { data: apps })
```
Comment (mcdonnelldean) [inline server/plugins/concorda-user/controller/application.js]: {data: apps} > {connected_services: connectedServices}

**Pull request merged**